cmd/autoscaler/main.go

```diff
@@ -181,22 +181,32 @@ func scalerConfig(logger *zap.SugaredLogger) *autoscaler.DynamicConfig {
 
 func uniScalerFactoryFunc(endpointsInformer corev1informers.EndpointsInformer) func(decider *autoscaler.Decider, dynamicConfig *autoscaler.DynamicConfig) (autoscaler.UniScaler, error) {
 	return func(decider *autoscaler.Decider, dynamicConfig *autoscaler.DynamicConfig) (autoscaler.UniScaler, error) {
-		if decider.Labels == nil {
+		if decider.Labels == nil || len(decider.Labels) == 0 {
 			return nil, fmt.Errorf("no labels set on Decider: %v", decider)
 		}
 
+		revName := decider.Labels[serving.RevisionLabelKey]
+		if revName == "" {
+			return nil, fmt.Errorf("no Revision label found in Decider: %v", decider)
+		}
+
+		serviceName := decider.Labels[serving.ServiceLabelKey]
+		if serviceName == "" {
+			return nil, fmt.Errorf("no Service label found in Decider: %v", decider)
+		}
+
+		configName := decider.Labels[serving.ConfigurationLabelKey]
+		if configName == "" {
+			return nil, fmt.Errorf("no Configuration label found in Decider: %v", decider)
+		}
```

Comment:
Should we implement a more common "check" method here, like:

```golang
func HasDefinedLabels(obj metav1.Object, labelKeys ...string) error {
	labels := obj.GetLabels()
	if len(labels) == 0 {
		return errors.New("no labels defined")
	}

	for _, key := range labelKeys {
		if labels[key] == "" {
			return fmt.Errorf("undefined label %q", key)
		}
	}
	return nil
}
```

Not sure if its worth it just for this instance though. If not, should we still implement a looping check here vs. repeating the code?

Reply:
Well, it makes the check slicker, but then we'll have to request the values separately in the call (or with vars, I'd got that way).